src/_guides/openapi/design-first-rails.md

```diff
@@ -0,0 +1,190 @@
+---
+title: Design-first API Rails Development
+authors: phil
+excerpt: Use OpenAPI to simplify building and testing Ruby on Rails APIs.
+---
+
+Ruby on Rails developers are blessed with a bunch of great [OpenAPI](https://spec.openapis.org/oas/latest.html) tooling, and can use either of the API Code-First workflow which was popular for a long time, or the follow the newer API Design-first workflow. 
+
+Instead of writing loads of code and sprinkling in some metadata later to create docs, the design-first workflow assumes you create the OpenAPI descriptions before writing any code at all. Once you have the OpenAPI description documents saved in your repository, you can leverage it at every step of the API lifecycle, producing mock APIs for clients to test assumptions with, produce client libraries without writing any code, make really effective contract testing, even generate backend code to get the application teams started once the contract is all signed off.
+
+This guide is going to look at two specific parts of the API design-first workflow and help you get them set up in Rails.
+
+1. Request Validation with OpenAPI
+2. Contract Testing in RSpec with OpenAPI
+
+Once you've defined your OpenAPI via [graphical editors](https://openapi.tools/#gui-editors), [text editors](https://openapi.tools/#text-editors), or [traffic sniffing](/guides/openapi/code-first/#traffic-sniffing), you can pop it into your Git repository somewhere like `api/openapi.yaml`. 
+
+Having this raw OpenAPI in your repository means that Bump.sh setup can actually come first, as there is no generation step with API Design first. To set up deployments to Bump.sh you can use the [CLI](https://github.com/bump-sh/cli#bump-deploy-file), [GitHub Actions](https://github.com/marketplace/actions/bump-sh#diff-on-pull-requests-only), or other [Continuous Integration](https://docs.bump.sh/help/continuous-integration/). Then as you progress through this guide implementing validation and contract testing, you will probaby notice mistakes in your OpenAPI, and they can be deployed with each commit/merge.
```

Comment:
> means that Bump.sh setup can actually come first

Since the Bump.sh part is now split/moved to the other article, this might need more context.

Reply:
Great point, I noticed a few other bits in there which felt lacking and padded them out too. dab6279